tools/cli/main.cpp

```diff
@@ -12,246 +12,859 @@
+                                                  TRAVERSE_LONG_OPT,
+                                                  FILE_DATATYPE,
+                                                  TRAVERSE_GLOSSARY1);
+static struct arg_file *proc_input_fs = arg_filen(NULL, NULL, FILE_DATATYPE, 1, MAX_NO_INPUTS, INPUT_GLOSSARY);
+static struct arg_end *proc_end = arg_end(ARG_MAX_ERRORS);
+static void *proc_argtable[] =
+        {proc, proc_out_f, proc_err_f, proc_output_type, proc_catalog_fs, proc_import_fs, proc_perf_f, proc_filter,
+         proc_traverse, proc_input_fs, proc_end};
+static int proc_nerrors;
+
+
+//  subcommand: compare
+//    ion compare [--output <file>]
+//    [--error-report <file>]
+//    [--output-format <type>]
+//    [--catalog <file>]...
+//    [--comparison-type <type>]
+//    ( - | <input_file>...)
+
+static struct arg_rex *comp = arg_rex1(NULL, NULL, COMPARE_COMMAND_NAME, NULL, REG_ICASE, NULL);
+
+static struct arg_file *comp_out_f = arg_file0(OUTPUT_SHORT_OPT,
+                                               OUTPUT_LONG_OPT, FILE_DATATYPE, OUTPUT_FILE_GLOSSARY);
+
+static struct arg_file *comp_err_f = arg_file0(ERR_SHORT_OPT, ERR_LONG_OPT,
+                                               FILE_DATATYPE, ERR_GLOSSARY);
+
+static struct arg_str *comp_output_type = arg_str0(OUT_TYPE_SHORT_OPT,
+                                                   OUT_TYPE_LONG_OPT,
+                                                   TYPE_DATATYPE,
+                                                   OUT_TYPE_GLOSSARY1);
+
+static struct arg_file *comp_catalog_fs = arg_filen(CATALOG_SHORT_OPT,
+                                                    CATALOG_LONG_OPT,
+                                                    FILE_DATATYPE,
+                                                    0,
+                                                    MAX_NO_INPUTS,
+                                                    CATALOG_GLOSSARY1);
+
+static struct arg_str *comp_comparison_type = arg_str0(COMPARE_TYPE_SHORT_OPT,
+                                                       COMPARE_TYPE_LONG_OPT,
+                                                       TYPE_DATATYPE,
+                                                       COMPARE_TYPE_GLOSSARY1);
+
+
+static struct arg_file *comp_input_fs = arg_filen(NULL, NULL, FILE_DATATYPE, 0, MAX_NO_INPUTS, INPUT_GLOSSARY);
+
+static struct arg_end *comp_end = arg_end(ARG_MAX_ERRORS);
+
+static void *comp_argtable[] =
+        {comp, comp_out_f, comp_err_f, comp_output_type, comp_catalog_fs, comp_comparison_type, comp_input_fs,
+         comp_end};
+
+static int comp_nerrors;
+
+//    subcommand: extract
+//    ion extract [--output <file>]
+//    [--error-report <file>]
+//    [--output-format <type>]
+//    (--symtab-name <name>)
+//    (--symtab-version <version>)
+//    ( - | <input_file>...)
+
+static struct arg_rex *xtract = arg_rex1(NULL, NULL, EXTRACT_COMMAND_NAME, NULL, REG_ICASE, NULL);
```

Comment:
Any reason not to just name all these "extract" vs "xtract"?

Reply:
Fat fingers on my part.